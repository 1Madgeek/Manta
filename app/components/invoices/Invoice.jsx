// Electron libs
const ipc = require('electron').ipcRenderer;

// React Libraries
import React, {Component} from 'react';
import PropTypes from 'prop-types';

// 3rd Party Libs
const format = require('date-fns/format');
const moment = require('moment');
const _ = require('lodash');

// Custom Component
import {TR, TD} from '../shared/Table';

// Component
class Invoice extends Component {
  constructor(props) {
    super(props);
    this.deleteInvoice = this.deleteInvoice.bind(this);
    this.previewInvoice = this.previewInvoice.bind(this);
  }

  shouldComponentUpdate(nextProps) {
    return this.props.data._id !== nextProps.data._id;
  }

  deleteInvoice() {
    const {data, deleteInvoice} = this.props;
    deleteInvoice(data._id);
  }

  previewInvoice() {
    ipc.send('preview-invoice', this.props.data);
  }

  // Render
  render() {
    const invoice = this.props.data;
    const {recipient} = invoice;
    return (
      <TR>
        <TD>
          {_.truncate(invoice._id, {
            length: 8,
            omission: '',
          })}
        </TD>
        <TD bold>
          {recipient.fullname}
        </TD>
        <TD>
          {invoice.dueDate
            ? moment(invoice.dueDate).format('DD/MM/YYYY')
            : '--'}
        </TD>
        <TD muted>
          {format(invoice.created_at, 'DD/MM/YYYY')}
        </TD>
        <TD bold success>
          {invoice.currency} {invoice.grandTotal}
        </TD>
        <TD actions>
          <a href="#" onClick={this.previewInvoice}>
            <i className="ion-search" />
          </a>
          <a href="#" onClick={this.deleteInvoice}>
            <i className="ion-android-cancel" />
          </a>
        </TD>
      </TR>
    );
  }
}

Invoice.propTypes = {
  data: PropTypes.object.isRequired,
  deleteInvoice: PropTypes.func.isRequired,
};

export default Invoice;
