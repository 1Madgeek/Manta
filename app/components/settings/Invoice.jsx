// Libraries
import React, { Component } from 'react';
import PropTypes from 'prop-types';
const ipc = require('electron').ipcRenderer;

// Custom Libs
const openDialog = require('../../renderers/dialog.js');
import _withFadeInAnimation from '../shared/hoc/_withFadeInAnimation';

import Currency from './_partials/invoice/Currency';
import Fields from './_partials/invoice/Fields';
import Other from './_partials/invoice/Other';
import Tax from './_partials/invoice/Tax';

// Component
class Invoice extends Component {
  constructor(props) {
    super(props);
    this.state = this.props.invoice;
    this.handleInputChange = this.handleInputChange.bind(this);
    this.handleTaxChange = this.handleTaxChange.bind(this);
    this.handleVisibilityChange = this.handleVisibilityChange.bind(this);
  }

  componentDidMount() {
    ipc.on('no-access-directory', (event, message) => {
      openDialog({
        type: 'warning',
        title: 'No Access Permisison',
        message: `${message}. Please choose a different directory!`,
      });
    });

    ipc.on('confirmed-export-directory', (event, path) => {
      this.setState({ exportDir: path }, () => {
        this.props.updateSettings('invoice', this.state);
      });
    });
  }

  componentWillUnmount() {
    ipc.removeAllListeners('no-access-directory');
    ipc.removeAllListeners('confirmed-export-directory');
  }

  handleInputChange(event) {
    const { setSavable } = this.props;
    const target = event.target;
    const value = target.type === 'checkbox' ? target.checked : target.value;
    const name = target.name;

    const canSave = this.canSave(name, value);
    setSavable(canSave);
    if (!canSave) {
      // Notifi
      console.log('Cant save');
    }

    this.setState({ [name]: value }, () => {
      this.props.updateSettings('invoice', this.state);
    });
  }

  canSave(ctrlName, value) {
    let valid = true;
    if (ctrlName === 'decimalFractions' && Number(value) < 0) {
      valid = false;
      openDialog({
        type: 'warning',
        title: 'Invalid decimal fraction setting',
        message: `${value} is an invalid for decimal fractions! Please correct setting to continue.`,
      });
    }

    return valid;
  }

  handleTaxChange(event) {
    const target = event.target;
    const name = target.name;
    const value = name === 'amount' ? parseFloat(target.value) : target.value;
    this.setState(
      {
        tax: Object.assign({}, this.state.tax, {
          [name]: value,
        }),
      },
      () => {
        this.props.updateSettings('invoice', this.state);
      }
    );
  }

  handleVisibilityChange(event) {
    const target = event.target;
    const name = target.name;
    const value = target.checked;
    this.setState(
      {
        required_fields: Object.assign({}, this.state.required_fields, {
          [name]: value,
        }),
      },
      () => {
        this.props.updateSettings('invoice', this.state);
      }
    );
  }

  selectExportDir() {
    ipc.send('select-export-directory');
  }

  render() {
    const {
      exportDir,
      template,
      currency,
      tax,
      required_fields,
      dateFormat,
      decimalFractions,
      currencyPlacement,
      decimalSeparator,
    } = this.state;
    return [
      <Fields
        key="required_fields_settings"
        required_fields={required_fields}
        handleVisibilityChange={this.handleVisibilityChange}
      />,
      <Tax
        key="tax_settings"
        handleTaxChange={this.handleTaxChange}
        tax={tax}
      />,
      <Currency
        key="currency_settings"
        currency={currency}
        handleInputChange={this.handleInputChange}
        decimalSeparator={decimalSeparator}
        currencyPlacement={currencyPlacement}
        decimalFractions={decimalFractions}
      />,
      <Other
        key="other_settings"
        dateFormat={dateFormat}
        exportDir={exportDir}
        template={template}
        handleInputChange={this.handleInputChange}
        selectExportDir={this.selectExportDir}
      />
    ];
  }
}

Invoice.propTypes = {
  invoice: PropTypes.object.isRequired,
  updateSettings: PropTypes.func.isRequired,
};

export default _withFadeInAnimation(Invoice);
