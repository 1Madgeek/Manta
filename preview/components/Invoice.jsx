// Libraries
import React, {Component} from 'react';
import PropTypes from 'prop-types';
const appConfig = require('electron').remote.require('electron-settings');

// Style
import styled from 'styled-components';

const Page = styled.div`
  position: relative;
  width: 21cm;
  height: 29.7cm;
  min-height: 29.7cm;
  min-width: 21cm;
  margin-left: auto;
  margin-right: auto;
  background: #FFFFFF;
  box-shadow: 0 0 10px rgba(0,0,0,.1);
  display: flex;
  border-radius: 4px;
}
`;

// Templates
import Minimal from '../templates/minimal';

// Component
class Invoice extends Component {
  shouldComponentUpdate(nextProps) {
    return this.props !== nextProps;
  }

  renderTemplate() {
    const {invoice, template, configs} = this.props;
    const props = {
      company: appConfig.get('info'),
      invoice,
      configs,
    };
    switch (template) {
      default: {
        return <Minimal {...props} />;
      }
    }
  }

  render() {
    return (
      <Page>
        { this.renderTemplate() }
      </Page>
    );
  }
}

Invoice.propTypes = {
  configs: PropTypes.object.isRequired,
  invoice: PropTypes.object.isRequired,
  template: PropTypes.string.isRequired,
};

export default Invoice;
