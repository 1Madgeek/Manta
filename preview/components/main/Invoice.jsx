// Libraries
import React, {Component} from 'react';
import PropTypes from 'prop-types';
const appConfig = require('electron').remote.require('electron-settings');

// Style
import styled from 'styled-components';

const Page = styled.div`
  position: relative;
  width: 21cm;
  height: 29.7cm;
  min-height: 29.7cm;
  min-width: 21cm;
  margin-left: auto;
  margin-right: auto;
  background: #FFFFFF;
  box-shadow: 0 0 10px rgba(0,0,0,.1);
  display: flex;
  border-radius: 4px;
}
`;

// Templates
import Minimal from '../../templates/minimal';
import Business from '../../templates/business';

// Component
class Invoice extends Component {
  shouldComponentUpdate(nextProps) {
    return this.props !== nextProps;
  }

  renderTemplate() {
    const {invoice, configs} = this.props;
    const props = {
      company: appConfig.get('profile'),
      invoice,
      configs,
    };
    switch (configs.template) {
      case 'business': {
        return <Business {...props} />;
      }
      default: {
        return <Minimal {...props} />;
      }
    }
  }

  render() {
    return (
      <Page>
        { this.renderTemplate() }
      </Page>
    );
  }
}

Invoice.propTypes = {
  configs: PropTypes.object.isRequired,
  invoice: PropTypes.object.isRequired,
};

export default Invoice;
