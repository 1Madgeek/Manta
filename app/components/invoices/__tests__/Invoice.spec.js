import Invoice from '../Invoice';
it('placeholder');
