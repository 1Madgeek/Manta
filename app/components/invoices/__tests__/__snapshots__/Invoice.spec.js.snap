// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly to the DOM matches snapshot 1`] = `
<div
  className="col-lg-6"
>
  <div
    className="Invoice__Wrapper-iBPCPb hxnMLw"
  >
    <div
      className="Invoice__StatusBar-gmZFuj aoiDg"
    />
    <div
      className="Invoice__Header-cAhbPy loejIf"
    >
      <div
        className="Invoice__Status-eKPgUx eiULaG"
      >
        <span>
          <i
            className="ion-loop"
          />
          Pending
        </span>
      </div>
      <button
        className="Button__ButtonLinkStyle-ioTbaT eFjcaV"
        onClick={[Function]}
      >
        <i
          className="ion-trash-a"
        />
      </button>
    </div>
    <div
      className="Invoice__Body-dSsrWx qPyCt"
    >
      <div
        className="Invoice__Row-gNlmug hPTbEG"
      >
        <div
          className="Invoice__Field-iXxbne dvBopU"
        >
          <label>
            Client
          </label>
          <h2>
            Penelope Hettinger
          </h2>
        </div>
      </div>
      <div
        className="Invoice__Row-gNlmug hPTbEG"
      >
        <div
          className="Invoice__Field-iXxbne dvBopU"
        >
          <label>
            Invoice Id
          </label>
          <p>
            37b2804e
          </p>
        </div>
        <div
          className="Invoice__Field-iXxbne dvBopU"
        >
          <label>
            Total Value
          </label>
          <p>
            USD
             
            3,843
          </p>
        </div>
      </div>
      <div
        className="Invoice__Row-gNlmug hPTbEG"
      >
        <div
          className="Invoice__Field-iXxbne dvBopU"
        >
          <label>
            Created On
          </label>
          <p>
            Invalid Date
          </p>
        </div>
        <div
          className="Invoice__Field-iXxbne dvBopU"
        >
          <label>
            Due Date
          </label>
          <p>
            --
          </p>
        </div>
      </div>
    </div>
    <div
      className="Invoice__Footer-imOzTt bLNPOf"
    >
      <div>
        <div
          className="SplitButton__Wrapper-dWaGGO iZVhhz"
        >
          <div
            className="SplitButton__ButtonsGroup-gPFJVZ jhIoxD"
          >
            <a
              href="#"
              onClick={[Function]}
            >
              Mark As Paid
            </a>
            <a
              href="#"
              onClick={[Function]}
            >
              <i
                className="ion-arrow-down-b"
              />
            </a>
          </div>
        </div>
      </div>
      <button
        className="Button__ButtonStyle-cORGTZ bzyUmu"
        onClick={[Function]}
      >
        Edit
      </button>
      <button
        className="Button__ButtonStyle-cORGTZ bzyUmu"
        onClick={[Function]}
      >
        View
      </button>
    </div>
  </div>
</div>
`;
